include/termcolor/termcolor.hpp

```diff
@@ -53,214 +52,264 @@ namespace termcolor
     {
         inline int colorize_index();
         inline FILE* get_standard_stream(const std::ostream& stream);
-        inline bool is_colorized(std::ostream& stream);
-        inline bool is_atty(const std::ostream& stream);
+        inline FILE* get_standard_stream(const std::wostream& stream);
+        template <typename CharT>
+        inline bool is_colorized(std::basic_ostream<CharT>& stream);
+        template <typename CharT>
+        inline bool is_atty(const std::basic_ostream<CharT>& stream);
 
     #if defined(TERMCOLOR_TARGET_WINDOWS)
-        inline void win_change_attributes(std::ostream& stream, int foreground, int background=-1);
+        inline void win_change_attributes(std::ostream& stream, int foreground, int background = -1);
+        inline void win_change_attributes(std::wostream& stream, int foreground, int background = -1);
     #endif
     }
 
+    template <typename CharT>
     inline
-    std::ostream& colorize(std::ostream& stream)
+    std::basic_ostream<CharT>& colorize(std::basic_ostream<CharT>& stream)
     {
         stream.iword(_internal::colorize_index()) = 1L;
         return stream;
     }
 
+    template <typename CharT>
     inline
-    std::ostream& nocolorize(std::ostream& stream)
+    std::basic_ostream<CharT>& nocolorize(std::basic_ostream<CharT>& stream)
     {
         stream.iword(_internal::colorize_index()) = 0L;
         return stream;
     }
 
+    template <typename CharT>
     inline
-    std::ostream& reset(std::ostream& stream)
+    std::basic_ostream<CharT>& reset(std::basic_ostream<CharT>& stream)
     {
         if (_internal::is_colorized(stream))
         {
         #if defined(TERMCOLOR_USE_ANSI_ESCAPE_SEQUENCES)
-            stream << "\033[00m";
+            if (std::is_same<CharT, wchar_t>::value)
```

Comment:
It seems there's no need for this `if`-branch since `basic_ostream<>` natively supoorts `char*` in addition to `CharT*`.

https://en.cppreference.com/w/cpp/io/basic_ostream/operator_ltlt2

Can you please remove these `if`-s here and below? It seems all we need to do is to use `std::basic_ostream<CharT>` instead of `std::ostream`.

Reply:
You can't use char string to wide char stream, here must be 'L' modifier before

Reply:
Wow, I had no idea that it would work without if else

Reply:
I didn’t even test it without if else, I was sure it wouldn’t work.